### ISSUE TITLE
$HOME not set after login

### ISSUE THREAD

**Issue by AdamChristiansen** [cylgom/ly#191]
I have two window managers (bspwm and berry) installed, which means ly presents the following login options:

- `berry`
- `bspwm`
- `xinitrc` (which only contains `exec bspwm`)
- `shell`

I use the fish shell, which gets its config from `$XDG_CONFIG_HOME/fish` (or `$HOME/.config/fish` if `$XDG_CONFIG_HOME` is not set). When I log in using any of the above 4 options, the screen flashes briefly and then returns to ly. I was able to record the screen on my phone and catch the message, which is:

```
error: Unable to locate the config directory.
error: Please set the XDF_CONFIG_HOME or HOME environment variable before starting fish.
```

So fish is telling me that `$HOME` is not set. If I change my default shell to bash I get the exact same issue when I try to log in, except there is no message. I'm assuming that bash is also unable to start because `$HOME` is undefined.

If I log in to a TTY normally, `$HOME` is set and it takes me straight to the default shell with no issues.

**Comment by AdamChristiansen** [comment]
I tried to do a little bit more digging into this issue. As a test, I modified `env_init()` in `login.c` to write a bunch of debug information to a file. This is the modified function:

```c
void env_init(struct passwd* pwd)
{
    extern char** environ;
    char* term = getenv("TERM");
    char* lang = getenv("LANG");
    // clean env
    environ[0] = NULL;

    #define LOGFILE(fmt, arg) \
        do { \
            FILE *f = fopen("/home/adam/ly-log.txt", "a"); \
            fprintf(f, (fmt), (arg)); \
            fclose(f); \
        } while (0)

    LOGFILE("pw_name:  %s\n", pwd->pw_name);
    LOGFILE("pw_dir:   %s\n", pwd->pw_dir);
    LOGFILE("pw_shell: %s\n", pwd->pw_shell);

    if (term != NULL)
    {
        setenv("TERM", term, 1);
    }
    else
    {
        setenv("TERM", "linux", 1);
    }
    LOGFILE("$TERM:    %s\n", getenv("TERM"));

    setenv("HOME", pwd->pw_dir, 1);     LOGFILE("$HOME:    %s\n", getenv("HOME"));
    setenv("PWD", pwd->pw_dir, 1);      LOGFILE("$PWD:     %s\n", getenv("PWD"));
    setenv("SHELL", pwd->pw_shell, 1);  LOGFILE("$SHELL:   %s\n", getenv("SHELL"));
    setenv("USER", pwd->pw_name, 1);    LOGFILE("$USER:    %s\n", getenv("USER"));
    setenv("LOGNAME", pwd->pw_name, 1); LOGFILE("$LOGNAME: %s\n", getenv("LOGNAME"));
    setenv("LANG", lang, 1);            LOGFILE("$LANG:    %s\n", getenv("LANG"));

    // Set PATH if specified in the configuration
    if (strlen(config.path))
    {
        int ok = setenv("PATH", config.path, 1);

        if (ok != 0)
        {
            dgn_throw(DGN_PATH);
        }
    }
}
```

This is the file contents after installing the new `ly` binary, rebooting, and attempting to login:

```
pw_name:  adam
pw_dir:   /home/adam
pw_shell: /bin/fish
$TERM:    linux
$HOME:    /home/adam
$PWD:     /home/adam
$SHELL:   /bin/fish
$USER:    adam
$LOGNAME: adam
```

Everything looks good, and it appears that the environment variables are all set. However, I still have the exact same issue where once my shell starts it complains that `$HOME` is not set.

**Comment by AdamChristiansen** [comment]
This was fixed by #183.
